.github/workflows/e2e-cache.yml

```diff
@@ -0,0 +1,88 @@
+name: Validate cache
+on:
+  push:
+    branches:
+      - main
+      - releases/*
+    paths-ignore:
+      - '**.md'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  save:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, windows-latest, ubuntu-latest]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Run setup-java with the cache for gradle
+        uses: ./
+        id: setup-java
+        with:
+          distribution: 'adopt'
+          java-version: '11'
+          cache: gradle
+      - name: Create files to cache
+        run: |
+          cat <<EOF > build.gradle
+            plugins { id 'java' }
+            repositories { mavenCentral() }
+            dependencies { implementation 'org.codehaus.groovy:groovy:1.8.6' }
+            tasks.register('downloadDependencies') { doLast {
+              def total = configurations.compileClasspath.inject (0) { sum, file ->
+                sum + file.length()
+              }
+              println total
+            }}
+          EOF
+          gradle downloadDependencies
+          if [ ! -d ~/.gradle/caches ]; then
+            echo "::error::The ~/.gradle/caches directory does not exist unexpectedly"
+            exit 1
+          fi
+  restore:
+    runs-on: ubuntu-latest
+    needs: save
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Create build.gradle
+        run: |
+          cat <<EOF > build.gradle
+            plugins { id 'java' }
+            repositories { mavenCentral() }
+            dependencies { implementation 'org.codehaus.groovy:groovy:1.8.6' }
+            tasks.register('downloadDependencies') { doLast {
+              def total = configurations.compileClasspath.inject (0) { sum, file ->
+                sum + file.length()
+              }
+              println total
+            }}
+          EOF
+          if [ -d ~/.gradle/caches ]; then
+            echo "::error::The ~/.gradle/caches directory exists unexpectedly"
+            exit 1
+          fi
```

Comment:
Could you please remove this check.

Reply:
Hi, could you tell me your motivation bit more detail? I thought it's necessary to ensure the directory was created by the action. 🤔

Reply:
Hello @KengoTODA. It can be removed because the new job will be started with no `~/.gradle/caches` directory. It will be created after `setup-java` step

Reply:
OK I understood, I've removed this process. Thanks!